### PR TITLE
Add cluster details for list clusters API

### PR DESCRIPTION
**Pull request by jessetrinity**
Adds a `ClusterDetails` type and a method to get the list of clusters from the service.

### CODE REVIEW

**Review by IlyaBiryukov — DISMISSED** [review]

File: cs/src/Contracts/ClusterDetails.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+namespace Microsoft.VsSaaS.TunnelService.Contracts;
+
+/// <summary>
+/// Tunnel service cluster details.
+/// </summary>
+public class ClusterDetails {
```
Comment (jasongin) [inline cs/src/Contracts/ClusterDetails.cs:6]: I'd suggest an `AzureLocation` property that has the standard Azure location name.

File: cs/src/Contracts/ClusterDetails.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    /// <summary>
+    /// A cluster identifier based on its region.
+    /// </summary>
+    public string? ClusterId { get; set; }
+    /// <summary>
+    /// The cluster DNS host.
+    /// </summary>
+    public string? Host { get; set; }
```
Comment (jasongin) [inline cs/src/Contracts/ClusterDetails.cs:14]: How about `Hostname` so it is less easily confused with a tunnel "host"?

File: cs/src/Contracts/ClusterDetails.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+namespace Microsoft.VsSaaS.TunnelService.Contracts;
+
+/// <summary>
+/// Tunnel service cluster details.
+/// </summary>
+public class ClusterDetails {
+    /// <summary>
+    /// A cluster identifier based on its region.
+    /// </summary>
+    public string? ClusterId { get; set; }
```
Comment (jasongin) [inline cs/src/Contracts/ClusterDetails.cs:10]: Why is it nullable? It should always have a value.

(In some other contracts, some required properties are nullable to support partial updates. But there will not be an API for updating clusters.)

File: cs/src/Contracts/ClusterDetails.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+namespace Microsoft.VsSaaS.TunnelService.Contracts;
+
+/// <summary>
+/// Tunnel service cluster details.
```
Comment (jasongin) [inline cs/src/Contracts/ClusterDetails.cs:4]: Include a brief explanation of what a "cluster" is.

File: cs/src/Contracts/ClusterDetails.cs (hunk 2 above)
Comment (jasongin) [reply cs/src/Contracts/ClusterDetails.cs:14]: Or should it be a full `https://` URI?

Comment (jessetrinity) [reply cs/src/Contracts/ClusterDetails.cs:14]: I had it with https before which was nicer in the CLI, but I thought host name might be better for URI builders. But URI builders are probably happy taking either form anyway.

**Review by jfullerton44 — DISMISSED** [review]
Go code looks good

File: cs/src/Contracts/ClusterDetails.cs (hunk 2 above)
Comment (jessetrinity) [reply cs/src/Contracts/ClusterDetails.cs]: Updated to use full uri.

**Review by IlyaBiryukov — APPROVED** [review]

**Review by jasongin — APPROVED** [review]

**Pull request merged**